### ISSUE TITLE
Vim9: status line legacy function wrongly evaluated as Vim9 function

### ISSUE THREAD

**Issue by lacygoill** [vim/vim#6698]
**Describe the bug**

Sometimes, a status line function written in Vim script legacy is wrongly evaluated as if it was written in Vim9 script.

**To Reproduce**

Run this shell command:

    vim -Nu NONE -S <(cat <<'EOF'
        vim9script

        set ls=2 stl=%!Stl()
        fu g:Stl()
            return '%{"x" is# "x"}'
        endfu

        s/^//c
    EOF
    )

`E15` is raised:

    E15: Invalid expression: is# "x"

That's because `is#` is disallowed in Vim9 script.

If you missed the error, press `q` to quit substituting, then run `:mess`.

**Expected behavior**

No error is raised, because:

   - `is#` is written in a legacy `:fu` function
   - `is#` is allowed in Vim script legacy

**Environment**

 - Vim version: 8.2 Included patches: 1-1434
 - OS: Ubuntu 16.04.7 LTS
 - Terminal: XTerm(358)

**Additional context**

The issue disappears if the substitution command is run from a legacy function:
```vim
vim9script
set ls=2 stl=%!Stl()
fu g:Stl()
    return '%{"x" is# "x"}'
endfu
fu Func()
    s/^//c
endfu
Func()
```
Or if the status line expression is directly included inside the `'statusline'` option:
```vim
vim9script
set ls=2 stl=%{'x'is#'x'}
fu Func()
    s/^//c
endfu
Func()
```

**Comment by lacygoill** [comment]
The issue also disappears if we remove the confirmation flag from the substitution command:
```vim
vim9script
set ls=2 stl=%!Stl()
fu g:Stl()
    return '%{"x" is# "x"}'
endfu
s/^//
```

**Comment by lacygoill** [comment]
The issue is *not* due to the fact that the `'statusline'` option was set in a Vim9 script.  I can reproduce even in a legacy script, simply by moving the substitution command in a `:def` function:
```vim
set ls=2 stl=%!Stl()
fu g:Stl()
    return '%{"x" is# "x"}'
endfu

def Func()
    s/^//c
enddef
call Func()
```

**Comment by lacygoill** [comment]
Fixed by [8.2.1437](https://github.com/vim/vim/commit/c9edd6b58218d25bfc5389af901021dc0eb82578).
